codex-cli/src/utils/transcriber.ts

```diff
@@ -0,0 +1,214 @@
+import { OpenAIRealtimeWS } from 'openai/beta/realtime/ws';
+import { EventEmitter } from 'events';
+import OpenAI from 'openai';
+// workaround since pvrecorder-node is a commonjs module
+import { createRequire } from 'node:module';
+const require = createRequire(import.meta.url);
+const { PvRecorder } = require('@picovoice/pvrecorder-node'); 
+
+// API configuration
+const API_KEY = process.env['OPENAI_API_KEY'];
+
+export interface TranscriptionEvent {
+  type: string;
+  delta?: string;
+  transcript?: string;
+}
+
+export interface TranscriberOptions {
+  model?: string;
+  language?: string;
+}
+
+export class RealtimeTranscriber extends EventEmitter {
+  private rt: OpenAIRealtimeWS | null = null;
+  private recorder: typeof PvRecorder | null = null;
+  private isConnected = false;
+  private isRecording = false;
+  private model: string;
+  private language: string;
+
+  constructor(options: TranscriberOptions = {}) {
+    super();
+    this.model = options.model || 'gpt-4o-mini-transcribe';
+    this.language = options.language || 'en';
+    this.setupSignalHandlers();
+  }
+
+  private setupSignalHandlers() {
+    process.on('SIGINT', () => this.cleanup());
+    process.on('SIGTERM', () => this.cleanup());
+  }
+
+  public async start() {
+    try {
+      // Check API key
+      if (!API_KEY) {
+        throw new Error('OPENAI_API_KEY not found in environment variables');
+      }
+      
+      // Initialize OpenAI client
+      const client = new OpenAI({ apiKey: API_KEY });
```

Comment:
Do we need `baseUrl` and default headers for this client, like `AgentLoop` does?